gax-java/gax/src/main/java/com/google/api/gax/rpc/EndpointContext.java

```diff
@@ -288,6 +303,36 @@ private String determineEndpoint() throws IOException {
       return endpoint;
     }
 
+    /** Determine if S2A can be used */
+    @VisibleForTesting
+    boolean shouldUseS2A() {
+      // If EXPERIMENTAL_GOOGLE_API_USE_S2A is not set to true, skip S2A.
+      String s2AEnv;
+      if (envProvider() != null) {
+        s2AEnv = envProvider().getenv(S2A_ENV_ENABLE_USE_S2A);
+      } else {
+        s2AEnv = System.getenv(S2A_ENV_ENABLE_USE_S2A);
+      }
```

Comment:
I think we may be able to just do 
`String s2AEnv = envProvider().getEnv(...)`

If we modify newBuilder to use the System.getEnv() call by default:
```
  public static Builder newBuilder() {
    return new AutoValue_EndpointContext.Builder()
        .setSwitchToMtlsEndpointAllowed(false)
        .setUsingGDCH(false)
        .setEnvProvider(System::getEnv);
  }
  ```

Let me know if that works or not.

Reply:
Thanks for pointing this out! Done in https://github.com/googleapis/sdk-platform-java/pull/3326/commits/257c51578b71d86043576a234480ce7029c11738